lambdas/check-existing-identity/src/main/java/uk/gov/di/ipv/core/checkexistingidentity/CheckExistingIdentityHandler.java

```diff
@@ -165,270 +169,261 @@ public Map<String, Object> handleRequest(JourneyRequest event, Context context)
-                                    LOG_MESSAGE_DESCRIPTION.getFieldName(),
-                                    "Returning CI error response.")
-                            .with(
-                                    LOG_ERROR_JOURNEY_RESPONSE.getFieldName(),
-                                    journeyResponse.toString());
-                    LOGGER.info(message);
-                });
-
-        return contraIndicatorJourneyResponse;
-    }
-
     @Tracing
-    private Optional<Map<String, Object>> getFaceToFaceResponse(
-            CriResponseItem faceToFaceRequest, VcStoreItem faceToFaceVc) {
-        if (Objects.isNull(faceToFaceVc) && !Objects.isNull(faceToFaceRequest)) {
-            final String requestStatus = faceToFaceRequest.getStatus();
-            if (requestStatus.equals(CriResponseService.STATUS_PENDING)) {
+    private Map<String, Object> buildF2FIncompleteResponse(CriResponseItem faceToFaceRequest) {
+        switch (faceToFaceRequest.getStatus()) {
+            case CriResponseService.STATUS_PENDING -> {
                 LOGGER.info(
                         new StringMapMessage()
                                 .with(
                                         LOG_MESSAGE_DESCRIPTION.getFieldName(),
                                         "F2F cri pending verification."));
-
-                return Optional.of(JOURNEY_PENDING);
+                return JOURNEY_PENDING;
             }
-            if (requestStatus.equals(CriResponseService.STATUS_ERROR)) {
+            case CriResponseService.STATUS_ERROR -> {
                 LOGGER.info(
                         new StringMapMessage()
                                 .with(LOG_MESSAGE_DESCRIPTION.getFieldName(), "F2F cri error"));
-                return Optional.of(JOURNEY_FAIL);
+                return JOURNEY_F2F_FAIL;
             }
+            default -> {
+                LOGGER.info(
+                        new StringMapMessage()
+                                .with(
+                                        LOG_MESSAGE_DESCRIPTION.getFieldName(),
+                                        "F2F unexpected status: " + faceToFaceRequest.getStatus()));
+                return JOURNEY_F2F_FAIL;
+            }
+        }
+    }
+
+    private Map<String, Object> buildCiBreachingResponse(ContraIndicators contraIndicators)
+            throws CiRetrievalException, UnrecognisedCiException, ConfigException {
+        JourneyResponse journeyResponse = JOURNEY_FAIL_WITH_CI;
+
+        var cimitConfig = configService.getCimitConfig();
+        for (var ci : contraIndicators.getContraIndicatorsMap().values()) {
+            if (cimitConfig.containsKey(ci.getCode()) && !ci.isMitigated()) {
+                journeyResponse = new JourneyResponse(cimitConfig.get(ci.getCode()));
+                break;
+            }
+        }
+
+        StringMapMessage message = new StringMapMessage();
+        message.with(LOG_MESSAGE_DESCRIPTION.getFieldName(), "Returning CI error response.")
```

Comment:
Without the logging this method could avoid having to reassign journeyResponse and instead just return a journeyResponse when appropriate